content/docs/usage/csi-driver-spiffe/README.md

```diff
@@ -76,16 +75,48 @@ The approver ensures that requests have:
    the CertificateRequest;
 5. a SPIFFE ID Trust Domain matching the one that was configured at startup.
 
-If any of these checks do not pass, the CertificateRequest will be marked as
-Denied, else it will be marked as Approved. The approver will only manage
-CertificateRequests who request from the same [IssuerRef](../../usage/certificaterequest.md)
-that has been configured.
+The approver only considers CertificateRequests which have the
+`spiffe.csi.cert-manager.io/identity` annotation, which is added by csi-driver-spiffe
+ to all requests it creates.
 
 ## Installation
 
 See the [installation guide](./installation.md) for instructions on how to
 install csi-driver-spiffe.
 
+## Security Considerations
+
+csi-driver-spiffe deals with highly valuable credentials which should be kept secret. The design
+of using a Kubernetes CSI volume makes it simple to restrict access to only the pod which mounts
+the CSI volume, but care should still be taken not to expose the private key created by csi-driver-spiffe.
+
+csi-driver-spiffe _always_ uses the token of the pod it's issuing for to create the CertificateRequest
+resource for the SVID it creates. That means that:
+
+1. During issuance, csi-driver-spiffe has the ability to do whatever the Pod can do; bear in mind that a
+   compromised csi-driver-spiffe pod could abuse these permissions, although in normal operation only the
+   CertificateRequest permission is used.
+2. Importantly: all pods using csi-driver-spiffe must have permission to create CertificateRequest resources.
+
+The requirement for a pod to have permission to create CertificateRequests has important security
+implications. If a Pod with these permissions is compromised it can create arbitrary CertificateRequest
+resources, which could reference arbitrary issuers in the cluster.
+
+Since csi-driver-spiffe requires that approval is used in-cluster, this risk is mitigated by approval.
+
+It's important, though, that any other forms of approval in the cluster (such as approver-policy) are
+carefully configured not to overlap with the csi-driver-spiffe approver and to restrict access to other
+issuers.
+
+For example, an approver-policy `CertificateRequestPolicy` resource which allowed any pod to issue
+using an ACME (Let's Encrypt) issuer might allow a compromised Pod to issue a publicly trusted
+certificate, if that Pod used csi-driver-spiffe and thus had permissions to create CertificateRequest
+resources.
+
+Using csi-driver-spiffe safely means considering which approval methods are available in your cluster
+and carefully configuring those approvers to ensure that pods cannot target issuers they're not meant
+to be able to use.
```

Comment:
Thanks, this section is super useful; I haven't found a design doc for csi-driver-spiffe, but what you wrote makes a good description of the design!